board/batocera/fsoverlay/etc/init.d/S99userservices

```diff
@@ -17,28 +17,38 @@ then
 fi
 
 # user services
-find /userdata/system/services -type f |
+find -L /userdata/system/services -type f |
     while read SERVICE
     do
-	BASE_SERVICE=${SERVICE##*/}
-	SRVVAR=__SERVICE__${BASE_SERVICE%.*}
-	if test "${!SRVVAR}" = 1
-	then
-	    bash "${SERVICE}" "${1}" &
-	fi
+    BASE_SERVICE=${SERVICE##*/}
+    if echo "${BASE_SERVICE,,}" | grep -q -E ^[a-z].[a-z0-9_]*$; then
```

Comment:
The commas seem extraneous, what's the purpose?

Reply:
- Using grep here is inefficient
- It's obfuscation to perform what should be an identical check in two scripts using completely different mechanisms
- This is a basic regexp so the `-E` flag is unnecessary
- The regexp has two bugs: it allows bad characters in the second position, and doesn't allow upper case characters

Reply:
test=YouKnow?; ${test^^} -> YOUKNOW?
test=YouKnow?; ${test,,} -> youknow?